### ISSUE TITLE
Microsoft.Compute/virtualMachines/extensions Resource Not Found

### ISSUE THREAD

**Issue by nckltcha** [grafana/azure-monitor-datasource#41]
I'm getting an error when trying to pull the additional metrics from the VM Extensions namespace:

esponse:Object
error:Object
code:"ResourceNotFound"
message:"The Resource 'Microsoft.Compute/virtualMachines/extensions' under resource group '****' was not found."

Doesn't look like a permission issue as Datadog was able to pull these using the same credentials.

**Comment by daniellee** [comment]
I'm seeing something similar. Can you see any metrics for vm extensions in the azure portal?

**Comment by bleep-io** [comment]
Yes I can see the extension metrics in the portal - I could also see them when using another tool - datadog.

![image](https://user-images.githubusercontent.com/1321647/37956730-0e926a58-31a4-11e8-8adc-6ba2e0881199.png)

**Comment by AlexandarY** [comment]
I think this and #46 are related. Diagnostics data is stored in Table storage and can be accessed only via Storage API.

**Comment by oxie** [comment]
Hello, i am having the same problem i am using azure grafana plugin to see metrics, but then i want to see disk space in / and for that i have installed LinuxDiagnostics to my VM  version 3.0, then from grafana i get Resource Not found. 
![screen shot 2018-06-22 at 16 55 52](https://user-images.githubusercontent.com/10477248/41783454-2d2ba640-763d-11e8-9bbe-974ba87cb64a.png)

How can i visualise that kind of metrics in grafana, is it possible at all?

**Comment by bleep-io** [comment]
I gave up on this extension and installed telegraf on the VMs to extract the metrics I need

**Comment by andreminelli** [comment]
> Hello, i am having the same problem i am using azure grafana plugin to see metrics, but then i want to see disk space in / and for that i have installed LinuxDiagnostics to my VM version 3.0
> 
> <img alt="screen shot 2018-06-22 at 22 26 09" width="677" src="https://user-images.githubusercontent.com/10477248/41797866-e5b72756-766b-11e8-8bc0-4fbef4f54971.png">
> 
> ![screen shot 2018-06-22 at 16 57 29](https://user-images.githubusercontent.com/10477248/41783559-66ec422c-763d-11e8-92b0-9c52814e20fe.png)
> 
> Then from grafana i get Resource Not found.
> 
> ![screen_shot_2018-06-22_at_16_55_52](https://user-images.githubusercontent.com/10477248/41793770-6dc28b90-765d-11e8-8b98-60160a1aeaec.png)
> 
> How can i visualise that kind of metrics in grafana, is it possible at all?

Any updates about this issue? I have the same problem :(

**Comment by oxie** [comment]
Sorry mate...

the guys seems focused on other issues... support seems very low... better
look for workaround bro... this aint getting done to work :/

Regards
Valentin

**Comment by bleep-io** [comment]
I just installed the telegraf client on the machine and collected my metrics that way

**Comment by daniellee** [comment]
@oxie 
>  i hate it how they promote grafana as the No1 Tool for
monitoring and then its so sucky at parts

Hey, this is open source - it didn't cost you a cent. The rest of your comment doesn't make much sense. Not sure who "they" is or what you mean by no db etc.

@bleep-io telegraf is a great solution - sounds like a very reliable way to collect metrics.

I never really got any good data in this thread to help me troubleshoot this. This stuff is totally undocumented.

I'm looking at a virtual machine with a couple of extensions:

![image](https://user-images.githubusercontent.com/434655/49408352-2aaf8d00-f75c-11e8-8bb2-61fae92684cc.png)

And I don't see any of those metrics in the Azure Portal which makes it hard to reverse engineer the api.
